src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -59,6 +61,38 @@ public static int getFileSizeInKB(String filePath) {
         return Math.toIntExact((file.length() / BYTES_PER_KILOBYTES) + 1L); // Add one so that integer division rounds up
     }
 
+    /**
+     * This method retrieves a specified field mapping from a set of mapping properties.
+     *
+     * @param properties A map representing properties, where each key is a property name and
+     *                   the value is either a sub-map of properties or the property value itself.
+     * @param fieldPaths The field path list that make up the path to the field mapping.
+     * @return           The value of the field if found, or null if the field is not present in the map.
+     */
+    public static Object getFieldMapping(final Map<String, Object> properties, final String[] fieldPaths) {
```

Comment:
why this function is public?

Reply:
@navneet1v I think we might use this method in other place and we can add unit test to this public method.

Reply:
When needed we can make this public, right now as there is no use case for that, we should not make this public.

Another thing, making a function public because we need to do unit testing is not a good reason. We should make sure we are testing all branches of this function via caller function.

Reply:
@navneet1v Changed it to private and added related unit tests from public function `validateKnnField`